### PR TITLE
GPC-NONE: Update test to include lambdas

### PR DESCRIPTION
**Pull request by oskarwilliams**

### CODE REVIEW

File: .github/workflows/application-pull-request.yml [hunk 1]
```diff
@@ -24,6 +24,8 @@ concurrency:
 jobs:
   test:
     uses: ./.github/workflows/test.yml
+    secrets:
+      SAM_APP_VALIDATE_ROLE_ARN: ${{ secrets.SAM_APP_VALIDATE_ROLE_ARN }}
```
Comment (olevett) [inline .github/workflows/application-pull-request.yml:28]: Is this really that sensitive? Or does it differ by env?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -24,28 +27,25 @@ jobs:
...
-      - name: Report
-        uses: dorny/test-reporter@v1
-        if: always()
-        with:
-          name: Test results
-          path: ${{ github.workspace }}/platform/build/test-results/test/TEST-*.xml
-          reporter: java-junit
-          only-summary: true
-
-      - name: Report test coverage
```
Comment (olevett) [inline .github/workflows/test.yml:38]: Do we not want test coverage?

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -24,28 +27,25 @@ jobs:
         uses: gradle/gradle-build-action@v2.6.0
 
       - name: Execute Gradle build
-        run: ./gradlew clean platform:test
+        run: ./gradlew clean test
 
-      - name: Report
```
Comment (olevett) [inline .github/workflows/test.yml:29]: @ethanmills  I think you added this test reporter, is it better/different to the mikepenz one?

**Review by olevett — APPROVED** [review]

File: .github/workflows/application-pull-request.yml (hunk 1 above)
Comment (oskarwilliams) [reply .github/workflows/application-pull-request.yml:28]: This is from the guide from DI, so I have kept it sensitive for their benefit

File: .github/workflows/test.yml (hunk 2 above)
Comment (oskarwilliams) [reply .github/workflows/test.yml:38]: We have that with a separate reporter, conforming with DI

Comment (oskarwilliams) [reply .github/workflows/test.yml:38]: Reverting this

File: .github/workflows/test.yml (hunk 3 above)
Comment (oskarwilliams) [reply .github/workflows/test.yml:29]: he's happy for us to try this

**Pull request merged**